File: src/lib/svelte/notes/service.ts
import * as repo from './repo';
import type { Note } from './db';

export type NoteInput = { title: string; body: string };

export const list = (db: D1Database): Promise<Note[]> => repo.list(db);
export const get = (db: D1Database, id: number) => repo.get(db, id);
export const create = (db: D1Database, input: NoteInput) => repo.create(db, input.title, input.body);
export const update = (db: D1Database, id: number, input: NoteInput) =>
	repo.update(db, id, input.title, input.body);
export const remove = (db: D1Database, id: number) => repo.remove(db, id);

File: src/lib/norns/notes/server/service.c
import type { NotesRepo } from './repo'

export type Note =
  id: number
  title: string
  body: string
  created_at: number
  updated_at: number

export type NoteInput =
  title: string
  body: string

export class NotesService
  repo: NotesRepo
  constructor(@repo: NotesRepo)

  list(): Promise<Note[]>
    @repo.list()

  get(id: number): Promise<Note | null>
    @repo.get id

  create(input: NoteInput): Promise<number>
    @repo.create input.title, input.body

  update(id: number, input: NoteInput): Promise<void>
    @repo.update id, input.title, input.body

  remove(id: number): Promise<void>
    @repo.remove id
